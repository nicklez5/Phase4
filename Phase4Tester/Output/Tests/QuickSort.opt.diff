--- SelfTestCases/QuickSort.opt.out	2011-04-28 14:55:21.000000000 -0700
+++ ./Output/Tests/QuickSort.opt.out	2020-03-14 13:04:38.053787052 -0700
@@ -1,23 +1,12 @@
-20
-7
-12
-18
-2
-11
-6
-9
-19
-5
-9999
-2
-5
-6
-7
-9
-11
-12
-18
-19
-20
-0
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/QuickSort.opt.compiler.out line 372 column 11: "0": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/QuickSort.opt.compiler.out line 386 column 11: "1": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/QuickSort.opt.compiler.out line 400 column 11: "2": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/QuickSort.opt.compiler.out line 414 column 11: "3": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/QuickSort.opt.compiler.out line 428 column 11: "4": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/QuickSort.opt.compiler.out line 442 column 11: "5": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/QuickSort.opt.compiler.out line 456 column 11: "6": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/QuickSort.opt.compiler.out line 470 column 11: "7": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/QuickSort.opt.compiler.out line 484 column 11: "8": operand is of incorrect type
+Error in /home/jackson/Desktop/phase4/Phase4Tester/./Output/Tests/QuickSort.opt.compiler.out line 498 column 11: "9": operand is of incorrect type
 
+Processing terminated due to errors.
