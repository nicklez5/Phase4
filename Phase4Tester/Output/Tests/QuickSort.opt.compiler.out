.data

empty_QS:

.text

  jal Main
  li $v0 10
  syscall

Main:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 8
  sw $ra -4($fp)
  li $a0 8
  jal _heapAlloc
  bnez $t0 null1
  la $a0 _str0
  j _error
null1:
  move $a0 $t0
  li $a1 10
  jal QS.Start
  move $t0 $v0
  move $a0 $t0
  jal _print
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 8
  jr $ra

QS.Start:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 12
  sw $ra -4($fp)
  sw $s0 0($sp)
  move $s0 $a0
  move $t0 $a1
  move $a0 $s0
  move $a1 $t0
  jal QS.Init
  move $a0 $s0
  jal QS.Print
  li $a0 9999
  jal _print
  lw $t0 4($s0)
  li $t9 1
  subu $t0 $t0 $t9
  move $a0 $s0
  li $a1 0
  move $a2 $t0
  jal QS.Sort
  move $a0 $s0
  jal QS.Print
  li $v0 0
  lw $s0 0($sp)
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 12
  jr $ra

QS.Sort:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 20
  sw $ra -4($fp)
  sw $s0 0($sp)
  sw $s1 4($sp)
  sw $s2 8($sp)
  move $s0 $a0
  move $t0 $a1
  move $s1 $a2
  li $t1 0
  slt $t2 $t0 $s1
  beqz $t2 if1_else
  lw $t2 0($s0)
  bnez $t2 null2
  la $a0 _str0
  j _error
null2:
  lw $t3 0($t2)
  move $a0 $s1
  move $a1 $t3
  bnez $t3 bounds1
  j _error
bounds1:
  mul $t3 $s1 4
  addu $t3 $t3 $t2
  lw $t3 4($t3)
  li $t9 1
  subu $s2 $t0 $t9
  move $t2 $s1
  li $t4 1
while1_top:
  beqz $t4 while1_end
  li $t5 1
while2_top:
  beqz $t5 while2_end
  addu $s2 $s2 1
  lw $t6 0($s0)
  bnez $t6 null3
  la $a0 _str0
  j _error
null3:
  lw $t7 0($t6)
  move $a0 $s2
  move $a1 $t7
  bnez $t7 bounds2
  j _error
bounds2:
  mul $t7 $s2 4
  addu $t7 $t7 $t6
  lw $t7 4($t7)
  slt $t6 $t7 $t3
  bnez $t6 if2_else
  li $t5 0
  j if2_end
if2_else:
  li $t5 1
if2_end:
  j while2_top
while2_end:
  li $t5 1
while3_top:
  beqz $t5 while3_end
  li $t9 1
  subu $t2 $t2 $t9
  lw $t6 0($s0)
  bnez $t6 null4
  la $a0 _str0
  j _error
null4:
  lw $t8 0($t6)
  move $a0 $t2
  move $a1 $t8
  bnez $t8 bounds3
  j _error
bounds3:
  mul $t8 $t2 4
  addu $t8 $t8 $t6
  lw $t7 4($t8)
  slt $t7 $t3 $t7
  bnez $t7 if3_else
  li $t5 0
  j if3_end
if3_else:
  li $t5 1
if3_end:
  j while3_top
while3_end:
  lw $t5 0($s0)
  bnez $t5 null5
  la $a0 _str0
  j _error
null5:
  lw $t7 0($t5)
  move $a0 $s2
  move $a1 $t7
  bnez $t7 bounds4
  j _error
bounds4:
  mul $t7 $s2 4
  addu $t7 $t7 $t5
  lw $t1 4($t7)
  lw $t7 0($s0)
  bnez $t7 null6
  la $a0 _str0
  j _error
null6:
  lw $t5 0($t7)
  move $a0 $s2
  move $a1 $t5
  bnez $t5 bounds5
  j _error
bounds5:
  mul $t5 $s2 4
  addu $t5 $t5 $t7
  lw $t7 0($s0)
  bnez $t7 null7
  la $a0 _str0
  j _error
null7:
  lw $t8 0($t7)
  move $a0 $t2
  move $a1 $t8
  bnez $t8 bounds6
  j _error
bounds6:
  mul $t8 $t2 4
  addu $t8 $t8 $t7
  lw $t8 4($t8)
  sw $t8 4($t5)
  lw $t8 0($s0)
  bnez $t8 null8
  la $a0 _str0
  j _error
null8:
  lw $t5 0($t8)
  move $a0 $t2
  move $a1 $t5
  bnez $t5 bounds7
  j _error
bounds7:
  mul $t5 $t2 4
  addu $t5 $t5 $t8
  sw $t1 4($t5)
  addu $t5 $s2 1
  slt $t5 $t2 $t5
  beqz $t5 if4_else
  li $t4 0
  j if4_end
if4_else:
  li $t4 1
if4_end:
  j while1_top
while1_end:
  lw $t4 0($s0)
  bnez $t4 null9
  la $a0 _str0
  j _error
null9:
  lw $t3 0($t4)
  move $a0 $t2
  move $a1 $t3
  bnez $t3 bounds8
  j _error
bounds8:
  mul $t3 $t2 4
  addu $t3 $t3 $t4
  lw $t4 0($s0)
  bnez $t4 null10
  la $a0 _str0
  j _error
null10:
  lw $t2 0($t4)
  move $a0 $s2
  move $a1 $t2
  bnez $t2 bounds9
  j _error
bounds9:
  mul $t2 $s2 4
  addu $t2 $t2 $t4
  lw $t2 4($t2)
  sw $t2 4($t3)
  lw $t2 0($s0)
  bnez $t2 null11
  la $a0 _str0
  j _error
null11:
  lw $t3 0($t2)
  move $a0 $s2
  move $a1 $t3
  bnez $t3 bounds10
  j _error
bounds10:
  mul $t3 $s2 4
  addu $t3 $t3 $t2
  lw $t2 0($s0)
  bnez $t2 null12
  la $a0 _str0
  j _error
null12:
  lw $t4 0($t2)
  move $a0 $s1
  move $a1 $t4
  bnez $t4 bounds11
  j _error
bounds11:
  mul $t4 $s1 4
  addu $t4 $t4 $t2
  lw $t4 4($t4)
  sw $t4 4($t3)
  lw $t4 0($s0)
  bnez $t4 null13
  la $a0 _str0
  j _error
null13:
  lw $t3 0($t4)
  move $a0 $s1
  move $a1 $t3
  bnez $t3 bounds12
  j _error
bounds12:
  mul $t3 $s1 4
  addu $t3 $t3 $t4
  sw $t1 4($t3)
  li $t9 1
  subu $t3 $s2 $t9
  move $a0 $s0
  move $a1 $t0
  move $a2 $t3
  jal QS.Sort
  addu $t3 $s2 1
  move $a0 $s0
  move $a1 $t3
  move $a2 $s1
  jal QS.Sort
  j if1_end
if1_else:
if1_end:
  li $v0 0
  lw $s0 0($sp)
  lw $s1 4($sp)
  lw $s2 8($sp)
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 20
  jr $ra

QS.Print:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 8
  sw $ra -4($fp)
  move $t0 $a0
  li $t1 0
while4_top:
  lw $t2 4($t0)
  slt $t2 $t1 $t2
  beqz $t2 while4_end
  lw $t2 0($t0)
  bnez $t2 null14
  la $a0 _str0
  j _error
null14:
  lw $t3 0($t2)
  move $a0 $t1
  move $a1 $t3
  bnez $t3 bounds13
  j _error
bounds13:
  mul $t3 $t1 4
  addu $t3 $t3 $t2
  lw $t3 4($t3)
  move $a0 $t3
  jal _print
  addu $t1 $t1 1
  j while4_top
while4_end:
  li $v0 0
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 8
  jr $ra

QS.Init:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 12
  sw $ra -4($fp)
  sw $s0 0($sp)
  move $s0 $a0
  move $t0 $a1
  sw $t0 4($s0)
  move $a0 $t0
  jal AllocArray
  move $t0 $v0
  sw $t0 0($s0)
  lw $t0 0($s0)
  bnez $t0 null15
  la $a0 _str0
  j _error
null15:
  lw $t1 0($t0)
  li $a0 0
  move $a1 $t1
  bnez $t1 bounds14
  j _error
bounds14:
  mul $t1 0 4
  addu $t1 $t1 $t0
  sw $20 4($t1)
  lw $t1 0($s0)
  bnez $t1 null16
  la $a0 _str0
  j _error
null16:
  lw $t0 0($t1)
  li $a0 1
  move $a1 $t0
  bnez $t0 bounds15
  j _error
bounds15:
  mul $t0 1 4
  addu $t0 $t0 $t1
  sw $7 4($t0)
  lw $t0 0($s0)
  bnez $t0 null17
  la $a0 _str0
  j _error
null17:
  lw $t1 0($t0)
  li $a0 2
  move $a1 $t1
  bnez $t1 bounds16
  j _error
bounds16:
  mul $t1 2 4
  addu $t1 $t1 $t0
  sw $12 4($t1)
  lw $t1 0($s0)
  bnez $t1 null18
  la $a0 _str0
  j _error
null18:
  lw $t0 0($t1)
  li $a0 3
  move $a1 $t0
  bnez $t0 bounds17
  j _error
bounds17:
  mul $t0 3 4
  addu $t0 $t0 $t1
  sw $18 4($t0)
  lw $t0 0($s0)
  bnez $t0 null19
  la $a0 _str0
  j _error
null19:
  lw $t1 0($t0)
  li $a0 4
  move $a1 $t1
  bnez $t1 bounds18
  j _error
bounds18:
  mul $t1 4 4
  addu $t1 $t1 $t0
  sw $2 4($t1)
  lw $t1 0($s0)
  bnez $t1 null20
  la $a0 _str0
  j _error
null20:
  lw $t0 0($t1)
  li $a0 5
  move $a1 $t0
  bnez $t0 bounds19
  j _error
bounds19:
  mul $t0 5 4
  addu $t0 $t0 $t1
  sw $11 4($t0)
  lw $t0 0($s0)
  bnez $t0 null21
  la $a0 _str0
  j _error
null21:
  lw $t1 0($t0)
  li $a0 6
  move $a1 $t1
  bnez $t1 bounds20
  j _error
bounds20:
  mul $t1 6 4
  addu $t1 $t1 $t0
  sw $6 4($t1)
  lw $t1 0($s0)
  bnez $t1 null22
  la $a0 _str0
  j _error
null22:
  lw $t0 0($t1)
  li $a0 7
  move $a1 $t0
  bnez $t0 bounds21
  j _error
bounds21:
  mul $t0 7 4
  addu $t0 $t0 $t1
  sw $9 4($t0)
  lw $t0 0($s0)
  bnez $t0 null23
  la $a0 _str0
  j _error
null23:
  lw $t1 0($t0)
  li $a0 8
  move $a1 $t1
  bnez $t1 bounds22
  j _error
bounds22:
  mul $t1 8 4
  addu $t1 $t1 $t0
  sw $19 4($t1)
  lw $t1 0($s0)
  bnez $t1 null24
  la $a0 _str0
  j _error
null24:
  lw $t0 0($t1)
  li $a0 9
  move $a1 $t0
  bnez $t0 bounds23
  j _error
bounds23:
  mul $t0 9 4
  addu $t0 $t0 $t1
  sw $5 4($t0)
  li $v0 0
  lw $s0 0($sp)
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 12
  jr $ra

AllocArray:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 8
  sw $ra -4($fp)
  move $t0 $a0
  mul $t1 $t0 4
  addu $t1 $t1 4
  move $a0 $t1
  jal _heapAlloc
  sw $t0 0($t1)
  move $v0 $t1
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 8
  jr $ra

_print:
  li $v0 1   # syscall: print integer
  syscall
  la $a0 _newline
  li $v0 4   # syscall: print string
  syscall
  jr $ra

_error:
  li $v0 4   # syscall: print string
  syscall
  li $v0 10  # syscall: exit
  syscall

_heapAlloc:
  li $v0 9   # syscall: sbrk
  syscall
  jr $ra

.data
.align 0
_newline: .asciiz "\n"
_str0: .asciiz "null pointer\n"
